src/MagentoDataLayerBase.ts

```diff
@@ -20,19 +20,30 @@ export abstract class MagentoDataLayerBase {
     name: EventName,
     handler: MagentoDataLayerEventHandler
   ): void {
+    const wrappedHandler = (event: EventName) => handler(event, this.mdl);
+
+    this.mdl._handlerMapper.set(handler, wrappedHandler);
```

Comment:
Shouldn't this map the `event` to the `wrappedHandler`? Or am I misunderstanding something?

```javascript
this.mdl._handlerMapper.set(event, wrappedHandler);
```

Reply:
It should map handler to wrappedHandler. An event can have many subscribers. When a user wants to unregister a handler they have to pass in the handler function they originally passed in so the ACDL knows which handler to unregister. The problem was the handler they passed in to add the event listener isn't the same one I pass on to the ACDL. So I have to maintain a mapping between the handler they pass in and the wrappedHandler I pass along.